rgf/test/test.py

```diff
@@ -73,6 +71,17 @@ def test_regressor(self):
         print("mse: " + str(mse))
         assert_less(mse, 6.0)
 
+    def test_sample_weight(self):
+        X_train, X_test, y_train, y_test = train_test_split(self.iris.data,
+                                                            self.iris.target,
+                                                            test_size=0.2,
+                                                            random_state=42)
+
+        clf = RGFClassifier()
+        y_pred = clf.fit(X_train, y_train).predict_proba(X_test)
+        y_pred_weighted = clf.fit(X_train, y_train, np.ones(y_train.shape[0])).predict_proba(X_test)
+        self.assertTrue(np.isclose(y_pred, y_pred_weighted).all())
```

Comment:
Code snippet as follows is one of the tests of the `sklearn.DecisionTreeClassifier`.

I think that it is a bug in RGF that sample_weight can not be set to zero.

```
import numpy as np

from sklearn.tree import DecisionTreeClassifier
from sklearn.utils.testing import assert_array_equal

X = np.arange(100)[:, np.newaxis]
y = np.ones(100)
y[:50] = 0.0

sample_weight = np.ones(100)
sample_weight[y == 0] = 0.0

clf = DecisionTreeClassifier(random_state=0)
clf.fit(X, y, sample_weight=sample_weight)
assert_array_equal(clf.predict(X), np.ones(100))
```